skills/hugging-face-model-trainer/scripts/eval_sft_adapter_example.py

```
@@ -0,0 +1,125 @@
+    if tokenizer.pad_token is None:
+        tokenizer.pad_token = tokenizer.eos_token
+
+    dataset = load_dataset(DATASET_NAME, DATASET_CONFIG, split="train")
+    dataset = dataset.shuffle(seed=SEED)
+
+    start = TRAIN_MAX_SAMPLES
+    end = min(start + EVAL_SAMPLES, len(dataset))
+    if end <= start:
+        raise ValueError(
+            "Held-out slice is empty. Reduce TRAIN_MAX_SAMPLES or EVAL_SAMPLES."
+        )
```

Ruff (0.15.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)